### PR TITLE
software-defined asset -> asset definition in docs

### PR DESCRIPTION
**Pull request by sryza**
## Summary & Motivation

This proposes using the term "asset definition" instead of "software-defined asset" in docs. We would continue to use the word "software-defined asset" in thought-leader-y content, to refer to the idea of managing data assets using software.

Reasoning:
- All Dagster definitions come from software. Calling it out in this particular case is inconsistent.
- "Software-defined asset" is a mouthful. This will matter even more in a world where we treat what we now call "external assets" as a kind of software-defined asset. Terms like "external software-defined asset" or "materializable software-defined asset" are unnecessarily long and difficult to work with.

### CODE REVIEW

File: docs/content/concepts/assets/software-defined-assets.mdx [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
-title: Software-defined assets | Dagster
-description: A software-defined asset is a description of how to compute the contents of a particular data asset.
+title: Asset definitions | Dagster
+description: A asset definition is a description of how to compute the contents of a particular data asset.
```
Comment (schrockn) [inline docs/content/concepts/assets/software-defined-assets.mdx:3]: An asset*

File: docs/content/concepts/assets/software-defined-assets.mdx [hunk 2]
```diff
@@ -14,20 +14,20 @@ description: A software-defined asset is a description of how to compute the con
   <a href="https://www.youtube.com/watch?v=lRwpcyd6w8k" target="new">
     demo
   </a>{" "}
-  videos to get a quick look at Software-defined assets.
+  videos to get a quick look at asset definitions.
 </Note>
 
-An **asset** is an object in persistent storage, such as a table, file, or persisted machine learning model. A **software-defined asset** is a description, in code, of an asset that should exist and how to produce and update that asset.
+An **asset** is an object in persistent storage, such as a table, file, or persisted machine learning model. A **asset definition** is a description, in code, of an asset that should exist and how to produce and update that asset.
```
Comment (schrockn) [inline docs/content/concepts/assets/software-defined-assets.mdx:20]: I think the final version of this is 

"A **asset definition** is a description, in code, of an asset that should exist."

And then we *add* how to produce and update it in the case where we add a materialization function. This would make "external assets" just asset definitions for assets that are materialized elsewhere.

**Review by schrockn — COMMENTED** [review]
Tl;dr is that I like it a lot

File: docs/content/concepts/assets/software-defined-assets.mdx (hunk 2 above)
Comment (sryza) [reply docs/content/concepts/assets/software-defined-assets.mdx:20]: Agree - the story behind this PR is that I started on the PR that makes that change, and thought it would be more reviewable if we made the "-> asset definition" change first.

**Review by erinkcochran87 — APPROVED** [review]
I also like this a lot. This allows us to more easily discuss different types of asset definitions (ex: multi, graph-backed, etc) without having to dance around including SDA language. Not necessarily in the current state of this page, but generally in our docs.

This is along the lines of what I'm thinking, but with some wordsmithing:

> A **asset definition** is a description, in code, of an asset that should exist and how to produce and update that asset... The `@asset` decorator is how you'll define basic assets, but depending on your use case, Dagster also offers other types of asset definitions:
> - Multi-assets <add detail>
> - Graph-backed assets <add detail>
> - ... etc

File: docs/content/concepts/assets/software-defined-assets.mdx (hunk 2 above)
Comment (schrockn) [reply docs/content/concepts/assets/software-defined-assets.mdx:20]: Awesome. Love where this is heading.

**Comment by sryza** [comment]
@schrockn @erinkcochran87 - I transformed this into a large PR that touches nearly all uses of "software-defined asset" on the docs site.
- I left the term in a couple places, e.g. where we talk about the "software-defined assets framework".
- I didn't do a bulk find/replace – I manually looked at every instance and made sure the replacement made sense.

If it would be helpful for review, happy to break this up into a few smaller PRs - e.g. one for concepts, one for integrations, one for tutorial.

File: docs/content/concepts/assets/graph-backed-assets.mdx [hunk 3]
```diff
@@ -59,7 +59,7 @@ def slack_files_table():
 
 ### Defining managed-loading dependencies for graph-backed assets
 
-Similar to software-defined assets, Dagster infers the upstream assets from the names of the arguments to the decorated function. Dagster will then delegate loading the data to an [I/O manager](/concepts/io-management/io-managers).
+Similar to single-op asset definitions, Dagster infers the upstream assets from the names of the arguments to the decorated function. Dagster will then delegate loading the data to an [I/O manager](/concepts/io-management/io-managers).
```
Comment (schrockn) [inline docs/content/concepts/assets/graph-backed-assets.mdx:62]: op-backed?

File: docs/content/concepts/resources.mdx [hunk 4]
```diff
@@ -44,7 +44,7 @@ So, why use resources?
...
 | <PyObject object="build_init_resource_context"/> | Function for building an <PyObject object="InitResourceContext"/> outside of execution, intended to be used when testing a resource.                                                                                                    |
 | <PyObject object="build_resources"/>             | Function for initializing a set of resources outside of the context of a job's execution.                                                                                                                                               |
-| <PyObject object="with_resources"/>              | Advanced API for providing resources to a specific set of Software-defined Assets and source assets, overriding those provided to <PyObject object="Definitions"/>.                                                                     |
+| <PyObject object="with_resources"/>              | Advanced API for providing resources to a specific set of asset definitions, overriding those provided to <PyObject object="Definitions"/>.                                                                     |
```
Comment (schrockn) [inline docs/content/concepts/resources.mdx:47]: ❤️

**Review by schrockn — APPROVED** [review]
This is _fantastic_. I just skimmed and will leave it to @erinkcochran87 to do a in depth pass, but IMO this is a huge step forward for how we communicate about the system.
